src/connectionconfig/connectionconfig.ts

```diff
@@ -66,17 +66,18 @@ export class ConnectionConfig implements IConnectionConfig {
      * and next alphabetically by profile/server name.
      */
     public getConnections(getWorkspaceConnections: boolean): IConnectionProfile[] {
-        let profiles = [];
+        let profiles: IConnectionProfile[] = [];
         let compareProfileFunc = (a, b) => {
-            // Sort by profile name if available, otherwise fall back to server name
-            let nameA = a.profileName ? a.profileName : a.server;
-            let nameB = b.profileName ? b.profileName : b.server;
+            // Sort by profile name if available, otherwise fall back to server name or connection string
+            let nameA = a.profileName ? a.profileName : a.server ? a.server : a.connectionString;
+            let nameB = b.profileName ? b.profileName : b.server ? b.server : b.connectionString;
```

Comment:
Potentially either add in braces or have the result of (a.server ? a.server : a.connectionString) stored in a separate variable to make this more easily readable